templates/vue-demo-store/pages/checkout/index.vue

```diff
@@ -268,6 +281,42 @@ async function invokeLogout() {
 
 const loginModalController = useModal();
 const addAddressModalController = useModal();
+
+const refreshShoppingAndPaymentMethods = (params?: {
+  skipShipping?: boolean;
+  skipPayment?: boolean;
+}) => {
+  const promises = [
+    !params?.skipPayment ? getPaymentMethods({ forceReload: true }) : null,
+    !isVirtualCart.value || !params?.skipShipping
+      ? getShippingMethods({ forceReload: true })
+      : null,
+  ];
+
+  if (!isVirtualCart.value) isLoading["shippingMethods"] = true;
+  isLoading["paymentMethods"] = true;
+  return Promise.any(promises).finally(() => {
+    isLoading["shippingMethods"] = false;
+    isLoading["paymentMethods"] = false;
+  });
+};
```

Comment:
Just for easier readability. But not tested (!). So this is just a suggestion to think about readability.

```suggestion
const REFRESH_OPTIONS = {
    FORCE_RELOAD: { forceReload: true },
    SKIP_SHIPPING: 'skipShipping',
    SKIP_PAYMENT: 'skipPayment'
};

const refreshShoppingAndPaymentMethods = (params = {}) => {
    let { skipShipping, skipPayment } = params;

    let loadShipping = !isVirtualCart.value || !skipShipping;
    let loadPayment = !skipPayment;

    if (loadShipping) {
        isLoading["shippingMethods"] = true;
    }

    if (loadPayment) {
        isLoading["paymentMethods"] = true;
    }

    const promises = [
        loadPayment ? getPaymentMethods(REFRESH_OPTIONS.FORCE_RELOAD) : null,
        loadShipping ? getShippingMethods(REFRESH_OPTIONS.FORCE_RELOAD) : null,
    ];

    return Promise.any(promises).finally(() => {
        if (loadShipping) {
            isLoading["shippingMethods"] = false;
        }

        if (loadPayment) {
            isLoading["paymentMethods"] = false;
        }
    });
};
```

Reply:
I'd still go for two separate methods for readability, maintenance and DX

Reply:
I have chosen the one-method approach because all business logic is included in this function, like virtual product support and loaders. But I also don't think one function per method is wrong, so sure, why not, done.